### ISSUE TITLE
Mapping a RegularFileProperty from one task to a Property in another task in an included build fails on second run with configuration cache enabled

### ISSUE THREAD

**Issue by eygraber** [gradle/gradle#18522]
I have an included build that defines a plugin and some tasks. The plugin is applied by my Android project. If I have configuration cache enabled on the main project, then the second+ time I run the `bump` task, the build fails.

```
abstract class ShouldSkipTask : DefaultTask() {
  init {
    outputs.upToDateWhen { false }
  }

  @get:OutputFile
  abstract val shouldSkipProp: RegularFileProperty

  @TaskAction
  fun shouldSkip() {
    shouldSkipProp.get().asFile.writeText("true")
  }
}

abstract class BumpTask : DefaultTask() {
  init {
    outputs.upToDateWhen { false }
  }

  @get:Internal
  abstract val shouldSkipProp: Property<Boolean>

  @TaskAction
  fun bump() {
    if(shouldSkipProp.getOrElse(false)) {
      println("Skip")
    }
    else {
      println("Didn't Skip")
    }
  }
}

class IncludeBuildPlugin : Plugin<Project> {
    override fun apply(target: Project) {
        val skipTask = target.tasks.register("shouldSkip", ShouldSkipTask::class.java) {
            it.shouldSkipProp.set(target.layout.buildDirectory.file("should_skip.txt"))
        }

        target.tasks.register("bump", BumpTask::class.java) {
            // as a possibly related issue, if this dependsOn is removed, a different error occurs
           // Failed to query the value of task ':app:bump' property 'shouldSkipProp'.
           //   Querying the mapped value of task ':app:shouldSkip' property 'shouldSkipProp' before task ':app:shouldSkip' has completed is not supported

            it.dependsOn(skipTask)
            it.shouldSkipProp.set(
                skipTask.flatMap { skip ->
                    skip.shouldSkipProp.map { skipFile ->
                        skipFile.asFile.readText() == "true"
                    }
                }
            )
        }
    }
}
```

If I change `BumpTask.shouldSkipProp` to a `RegularFileProperty` and set it using:

```
            it.shouldSkipProp.set(
                skipTask.flatMap { skip ->
                    skip.shouldSkipProp
                }
            )
```

then the build works.

### Steps to Reproduce 
[Repro project](https://github.com/gradle/gradle/files/7274960/gradle-18522.zip)

Run `gradle :app:bump`

### Your Environment
Gradle 7.2

Build scan URL: Appears that because of the error, a scan is not generated

**Comment by bamboo** [comment]
This is no longer reproducible with 7.5-rc-3.

**Comment by bamboo** [comment]
Fixed in 7.5.
